NDTensors/src/lib/Sectors/src/category_product.jl

```diff
@@ -34,78 +52,224 @@ function Base.show(io::IO, s::CategoryProduct)
   return print(io, ")")
 end
 
-category_show(io::IO, k, v) = print(io, v)
-
+category_show(io::IO, ::Int, v) = print(io, v)
 category_show(io::IO, k::Symbol, v) = print(io, "($k=$v,)")
 
+function Base.isless(s1::C, s2::C) where {C<:CategoryProduct}
+  return categories_isless(categories(s1), categories(s2))
+end
+function Base.isless(s1::CategoryProduct, s2::CategoryProduct)
+  return categories_isless(categories(s1), categories(s2))
+end
+
+# =======================================  shared  =========================================
+# there are 2 implementations for CategoryProduct
+# - ordered-like with a Tuple
+# - dictionary-like with a NamedTuple
+
+# get clean results when mixing implementations
+categories_isequal(::Tuple, ::NamedTuple) = false
+categories_isequal(::NamedTuple, ::Tuple) = false
+
+categories_isless(::NamedTuple, ::Tuple) = throw(ArgumentError("Not implemented"))
+categories_isless(::Tuple, ::NamedTuple) = throw(ArgumentError("Not implemented"))
+
+categories_type(::Type{<:CategoryProduct{T}}) where {T} = T
+
+function categories_fusion_rule(cats1, cats2)
+  diff_cat = CategoryProduct(categories_diff(cats1, cats2))
+  shared1, shared2 = categories_common(cats1, cats2)
+  fused = map(fusion_rule, values(shared1), values(shared2))
+  factorized = factorize_gradedaxes(fused)
+  type_fixed = recover_category_product_type(typeof(shared1), factorized)
+  return type_fixed × diff_cat
+end
+
+# abelian case: there is no gradedaxis
+factorize_gradedaxes(fused::Tuple{Vararg{AbstractCategory}}) = fused
+
+# non-abelian case
+function factorize_gradedaxes(fused::Tuple)
+  # here fused contains at least one GradedOneTo
+  g0 = reduce(×, fused)
+  # convention: keep unsorted blocklabels as produced by F order loops in ×
+  return g0
+end
+
+function recover_category_product_type(T::Type, g0::AbstractGradedUnitRange)
+  new_labels = recover_category_product_type.(T, blocklabels(g0))
+  new_blocklengths = labelled.(unlabel.(blocklengths(g0)), new_labels)
+  return gradedrange(new_blocklengths)
+end
+
+function recover_category_product_type(T::Type, c::AbstractCategory)
+  return recover_category_product_type(T, CategoryProduct(c))
+end
+
+function recover_category_product_type(T::Type, c::CategoryProduct)
+  return recover_category_product_type(T, categories(c))
+end
+
+function recover_category_product_type(
+  T::Type{<:CategoryProduct}, cats::Tuple{Vararg{AbstractCategory}}
+)
+  return recover_category_product_type(categories_type(T), cats)
+end
+
+function recover_category_product_type(T::Type, cats::Tuple{Vararg{AbstractCategory}})
+  return CategoryProduct(T(cats))
+end
+
+sector(args...; kws...) = CategoryProduct(args...; kws...)
```

Comment:
I would vote for just removing `sector` for now, or at least not using it in the source code, and bringing it back later if we think it is needed. I find it clearer in the source code when I see `CategoryProduct` used since then I know what type is being created.